### PR TITLE
Fix DbConnectionAccessor Exception

### PR DESCRIPTION
**Pull request by mdockal**
Fixed DbConnectionAccessor to access ConnectionFactory.CreateConnection() instead of Session.DemandAsync().   The InvalidOperationException below from the existing DbTransaction was superseding underlying exceptions and not allowing a new DbTransaction to be created if needed.

Exception:
"InvalidOperationException: This SqliteTransaction has completed; it is no longer usable."

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Data/DbConnectionAccessor.cs [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
+            _store = store ?? throw new ArgumentNullException(nameof(store));
         }
 
         public async Task<IDbConnection> GetConnectionAsync()
         {
-            var transaction = await _session.DemandAsync();
-
-            return transaction.Connection;
+            var connection = _store.Configuration.ConnectionFactory.CreateConnection() as DbConnection;
+            await connection?.OpenAsync();
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Data/DbConnectionAccessor.cs]: Maybe the connection should be stored, and disposed when `IDbConnectionAccessor` is disposed.

Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Data/DbConnectionAccessor.cs]: Also, what happens if the connection is already open (in the case we reuse the instance).

Comment (mdockal) [reply src/OrchardCore/OrchardCore.Data/DbConnectionAccessor.cs]: I always run it in a using block like this:
using (var connection = await _connectionAccessor.GetConnectionAsync())
{
   //Do something with data here.
}
But that doesn't mean every developer will do it the same.   I suppose store the connection in a field and I can make DbConnectionAccessor implement IDisposable and dispose of the connection when it is disposed.

**Comment by mdockal** [comment]
Implemented Interface IDisposable in IDbConnectionAccessor and added a field for DbConnection in DbConnectionAccessor so that it can be reused.

**Pull request merged**